server/parsec/components/memory/datamodel.py

```diff
@@ -395,3 +399,24 @@ class MemoryBlock:
     created_on: DateTime
     # None if not deleted
     deleted_on: DateTime | None = None
+
+
+@dataclass(slots=True)
+class MemoryShamirSetup:
+    # The actual data we want to recover.
+    # It is encrypted with `data_key` that is itself split into shares.
+    # This should contains a serialized `LocalDevice`
+    ciphered_data: bytes
+    # The token the claimer should provide to get access to `ciphered_data`.
+    # This token is split into shares, hence it acts as a proof the claimer
+    # asking for the `ciphered_data` had it identity confirmed by the recipients.
+    reveal_token: ShamirRevealToken
+    # The Shamir recovery setup provided as a `ShamirRecoveryBriefCertificate`.
+    # It contains the threshold for the quorum and the shares recipients.
+    # This field has a certain level of duplication with the "shares" below,
+    # but they are used for different things (we provide the encrypted share
+    # data only when needed)
+    brief: ShamirRecoveryBriefCertificate
+    # The shares provided as a `ShamirRecoveryShareCertificate` since
+    # each share is aimed at a specific recipient.
+    shares: dict[UserID, bytes]
```

Comment:
@vxgmichel will we provide "weight" to user share ?

In simpler term it's how many shares a user get, example:

I create 5 shares with a minimum of 3 shares,
I give 2 shares to Alice and 2 shares to Bob and the last one to John.

To reconstruct the secret, Alice provide its 2 shares and either John or Bob need to provide one.

Reply:
@FirelightFlagboy  Yes, the shares are weighted. As explained in the comment, each value in the dict is a serialized `ShamirRecoveryShareCertificate`. This share certificate contains a `ciphered_share` field that the user can decrypt and deserialize into a `ShamirRecoveryShareData`. This structure contains a `weighted_share` field which is a list of `ShamirShare`. The length of this list is the weight of the share.